src/api/pickAppResource.ts

```diff
@@ -3,18 +3,39 @@
 *  Licensed under the MIT License. See License.txt in the project root for license information.
 *--------------------------------------------------------------------------------------------*/
 
-import { AzExtTreeItem, ITreeItemPickerContext } from "@microsoft/vscode-azext-utils";
+import { AzExtResourceType, AzExtTreeItem, ContextValueFilter, getAzExtResourceType, ITreeItemPickerContext, PickTreeItemWithCompatibility } from "@microsoft/vscode-azext-utils";
 import { PickAppResourceOptions } from "@microsoft/vscode-azext-utils/hostapi";
 import { ext } from "../extensionVariables";
-import { SubscriptionTreeItem } from "../tree/SubscriptionTreeItem";
 
 export async function pickAppResource<T extends AzExtTreeItem>(context: ITreeItemPickerContext, options?: PickAppResourceOptions): Promise<T> {
-    const subscription: SubscriptionTreeItem = await ext.appResourceTree.showTreeItemPicker(SubscriptionTreeItem.contextValue, context);
-    const appResource = await subscription.pickAppResource(context, options);
+    return await PickTreeItemWithCompatibility.resource<T>(context, ext.v2.api.resources.azureResourceTreeDataProvider, {
+        resourceTypes: convertAppResourceFilterToAzExtResourceType(options?.filter),
+        childItemFilter: convertExpectedChildContextValueToContextValueFilter(options?.expectedChildContextValue)
+    });
+}
+
+function convertExpectedChildContextValueToContextValueFilter(expectedChildContextValue?: PickAppResourceOptions['expectedChildContextValue']): ContextValueFilter | undefined {
+    return expectedChildContextValue ? { include: expectedChildContextValue } : undefined
+}
 
-    if (options?.expectedChildContextValue) {
-        return ext.appResourceTree.showTreeItemPicker(options.expectedChildContextValue, context, appResource);
-    } else {
-        return appResource as unknown as T;
+function convertAppResourceFilterToAzExtResourceType(filter?: PickAppResourceOptions['filter']): AzExtResourceType[] | undefined {
+    if (!filter) {
+        return undefined;
     }
+
+    filter = Array.isArray(filter) ? filter : [filter];
+    return filterMap(filter, getAzExtResourceType);
+}
+
+function filterMap<T, TMapped>(source: T[], predicateMapper: (item: T, index: number) => TMapped | null | undefined): TMapped[] {
+    let index = 0;
+    return source.reduce<TMapped[]>((accumulator, current) => {
+        const mapped = predicateMapper(current, index++);
+        // handles null or undefined
+        // eslint-disable-next-line eqeqeq
+        if (mapped != null) {
```

Comment:
Does this work instead and remove the need for the ESLint suppression? Not sure what other falsy values might be problems or not.
```suggestion
        if (!!mapped) {
```

Otherwise I'd almost prefer
```suggestion
        if (mapped !== null && mapped !== undefined) {
```

Reply:
I like doing `mapped !== null && mapped !== undefined`.